File: server/app/models/runner.py
from typing import List, Literal, Optional
from pydantic import BaseModel, Field

class RunnerProfile(BaseModel):
    name: str
    age: int
    sex: Literal["male", "female", "other"]
    experience_level: Literal["beginner", "intermediate", "advanced"]
    weekly_mileage_km: float = Field(..., description="Current average weekly mileage in km")
    preferred_units: Literal["km", "mi"] = "km"
    available_days: List[Literal["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]]
    constraints: List[str] = Field(
        default_factory=list,
        description="Injuries, schedule constraints, etc",
    )

File: client/lib/types/runner.ts
import type { Sex, Units, Day, ISODateString } from "./common";

export interface RunnerProfile {
  name: string;
  age: number;
  sex: Sex;
  experience_level: "beginner" | "intermediate" | "advanced";
  weekly_mileage_km: number;
  preferred_units: Units;
  available_days: Day[];            // ["Mon", "Wed", ...]
  constraints?: string[];
}

export interface RecentRun {
  date: ISODateString;              // ISO date string
  distance_km: number;
  duration_min: number;
  avg_pace_min_per_km: number;
  notes?: string;
}
